### DISCUSSION TITLE
Download only 1 folder of project

### DISCUSSION THREAD

**Discussion by southrivertech** [crowdin/crowdin-cli#346] [Q&A]
Hi,
 I have a project with two folders

/Folder1
/Folder2

I want to use the CLI to download just folder2. When I use crowdin download, it tries to download both folder1 and folder2. Is there a 

crowdin download --folder "folder2"

**Comment by DimaYashchyshyn-zz** [comment]
Hi, you can replace folders with branches and then download translations only from a specified branch 
crowdin download -b {branch_name}

**Reply by southrivertech** [reply]
Thank you for the suggestion.

Is there any option for just downloading an individual folder without breaking my project out into Branches?

**Comment by andrii-bodnar** [comment]
Hi @southrivertech!
Could you please share your configuration file? I would only like to see the `files` node

**Reply by southrivertech** [reply]
I do see that API has the ability to download a directory using....

https://api.crowdin.com/api/v2/projects/12345/directories/123

`#
# YML FILE
#
"project_identifier" : "MyProject"
"project_id" : "123456"
"api_key" : "fasdf12345"
"base_path" : "C:\\MyLocalRoot"
"preserve_hierarchy": true
files: [ {
 "source" : "\\Utils\\Source\\Properties\\Resources.resx",
  "translation" : "\\Utils\\Source\\Properties\\%file_name%.%two_letters_code%.%file_extension%"
}]
`

and in CrowdIn, MyProject will have multiple top level directories

/ProjectFolder1
/ProjectFolder2
/Utils

/ProjectFolder1/ and /ProjectFolder2/ are under a different base_path and are not peers of /Utils/ so locally I have 

C:/MyMainFolder/ProjectFolder1/
C:/MyMainFolder/ProjectFolder2/
C:/MyLocalRoot/Utils/

my crowdin.yml lives under C:/MyLocalRoot/CrowdIn/crowdin.yml  tries to download /ProjectFolder1 and /ProjectFolder2 and causes errors becase they have a different base_path

the same is true for my C:/MyMainFolder/Crowdin/crowdin.yml which only has Files for /ProjectFolder1 and /ProjectFolder2

there should be a crowdin download --Folder 'utils'   option to grab only 1 folder from the cloud and bring it down

**Reply by andrii-bodnar** [reply]
Thanks for the details provided!

The possible solution - is to split your configuration file by each project.
Each configuration file should contain only the respective source files pattern. 
Then, using `--config` parameter, you can run CLI for each project and CLI will download from Crowdin only the files configured in the configuration file.

Also, I see you using CLI v2. This version is no longer supported and I recommend you to use the CLI v3 (`project_id`, `api_token` should be used instead `project_identifier`, `api_key` in the configuration file).

**Reply by southrivertech** [reply]
thanks; that sound logical. I'll give it a whirl.
